pilot/pkg/serviceregistry/kube/controller/pod.go

```diff
@@ -136,9 +142,26 @@ func (pc *PodCache) update(ip, key string) {
 	pc.podsByIP[ip] = key
 	pc.IPByPods[key] = ip
 
+	if endpointsToUpdate, f := pc.needResync[ip]; f {
+		delete(pc.needResync, ip)
```

Comment:
Do we need lock here also?

Reply:
There is already a lock from the caller of update I think

Reply:
Confirmed.